### ISSUE TITLE
Zipper inspect should show if you're at a subtree root

### ISSUE THREAD

**Issue by zachallaun** [doorgan/sourceror#169]
Right now, inspecting the top of a subtree looks the same as if you're at the root. Would aid with debugging if there was some indication that you're at the root of a subtree.

```elixir
alias Sourceror.Zipper, as: Z

code = """
block do
  one()
  two()
  three()
end
"""

code |> Sourceror.parse_string!() |> Z.zip() |> Z.search_pattern("two()") |> Z.subtree() |> IO.inspect()

# currently prints
#Sourceror.Zipper<
  #root
  {:two,
   [
     trailing_comments: [],
     leading_comments: [],
     end_of_expression: [newlines: 1, line: 3, column: 6],
     closing: [line: 3, column: 5],
     line: 3,
     column: 1
   ], []}
>

# would be nice to print
#Sourceror.Zipper<
  #subtree root
  {:two,
   [
     trailing_comments: [],
     leading_comments: [],
     end_of_expression: [newlines: 1, line: 3, column: 6],
     closing: [line: 3, column: 5],
     line: 3,
     column: 1
   ], []}
>
```

**Comment by zachdaniel** [comment]
Clever 👍

**Comment by doorgan** [comment]
This would be a welcome change 👍

**Comment by doorgan** [comment]
Merged with #167
